### ISSUE TITLE
Error in REPL keymap on shift-tab

### ISSUE THREAD

**Issue by tecosaur** [JuliaLang/julia#49435]
On Julia 1.9-rc2, I've found a few times that if I tab on `pkg> add [a few letters]` I can get this error message:

```
┌ Error: Error in the keymap
│   exception =
│    MethodError: no method matching setmodifiers!(::Pkg.REPLMode.PkgCompletionProvider, ::REPL.LineEdit.Modifiers)
│    
│    Closest candidates are:
│      setmodifiers!(::REPL.LineEdit.MIState, ::REPL.LineEdit.Modifiers)
│       @ REPL ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2047
│      setmodifiers!(::REPL.LineEdit.Prompt, ::REPL.LineEdit.Modifiers)
│       @ REPL ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2048
│      setmodifiers!(::REPL.REPLCompletionProvider, ::REPL.LineEdit.Modifiers)
│       @ REPL ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:510
│      ...
```

The stacktrace follows

```
│    Stacktrace:
│      [1] setmodifiers!(p::REPL.LineEdit.Prompt, m::REPL.LineEdit.Modifiers)
│        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2048
│      [2] setmodifiers!(s::REPL.LineEdit.MIState, m::REPL.LineEdit.Modifiers)
│        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2047
│      [3] shift_tab_completion
│        @ ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2319 [inlined]
│      [4] (::REPL.LineEdit.var"#115#171")(::REPL.LineEdit.MIState, ::Any, ::Vararg{Any})
│        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2359
│      [5] #invokelatest#2
│        @ ./essentials.jl:816 [inlined]
│      [6] invokelatest
│        @ ./essentials.jl:813 [inlined]
│      [7] (::REPL.LineEdit.var"#27#28"{REPL.LineEdit.var"#115#171", String})(s::Any, p::Any)
│        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:1603
│      [8] prompt!(term::REPL.Terminals.TextTerminal, prompt::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
│        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2740
│      [9] run_interface(terminal::REPL.Terminals.TextTerminal, m::REPL.LineEdit.ModalInterface, s::REPL.LineEdit.MIState)
│        @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2642
│     [10] run_frontend(repl::REPL.LineEditREPL, backend::REPL.REPLBackendRef)
│        @ REPL ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/REPL.jl:1298
│     [11] (::REPL.var"#62#68"{REPL.LineEditREPL, REPL.REPLBackendRef})()
│        @ REPL ./task.jl:514
└ @ REPL.LineEdit ~/.julia/juliaup/julia-1.9.0-rc2+0.x64.linux.gnu/share/julia/stdlib/v1.9/REPL/src/LineEdit.jl:2742
```

Looks like perhaps a bit of `LineEdit.jl` was modified but not everything updated.

**Comment by mbauman** [comment]
This reliably happens in both shell and pkg modes upon typing <kbd>shift</kbd><kbd>tab</kbd>

**Comment by LilithHafner** [comment]
Fixed by #50822
